src/lib.rs

```diff
@@ -135,6 +135,10 @@ pub struct Diagnostic {
 
     /// The diagnostic's message.
     pub message: String,
+
+    /// A number used to associate children diagnostics with their
+    /// parent. Can be omitted.
+    pub group: Option<u64>,
```

Comment:
Can you add `#[serde(skip_serializing_if "Option::is_none")]`? That way it won't even be serialized as null